### DISCUSSION TITLE
[Q&A] How to define the IP used for inventory?

### DISCUSSION THREAD

**Discussion by blackbasket** [glpi-project/glpi-agent#685] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi...

I try to inventory the FortiGate firewall using the GLPI agent 1.8, but it uses a wrong IP address.

During the discovery process all IPs assigned to the firewall ports are added to the management port.
![image](https://github.com/glpi-project/glpi-agent/assets/64146687/1ec52f31-41d8-4014-9d83-6dc258b0034e)

During the inventory process always the first IP address is used. Even, if I set a IP range with only one IP address (172.31.255.1) to inventory, this is ignored.
```
[Wed May 29 10:31:59 2024][debug] #1, full snmp scan of 10.254.253.254 with credentials 3
[Wed May 29 10:32:14 2024][error] #1, SNMP communication error: no response from host 10.254.253.254
[Wed May 29 10:32:14 2024][debug2] #1, [http client] sending message:
<?xml version="1.0" encoding="UTF-8"?>
<REQUEST>
  <CONTENT>
    <DEVICE>
      <ERROR>
        <ID>44</ID>
        <MESSAGE>SNMP communication error: no response from host 10.254.253.254</MESSAGE>
        <TYPE>NETWORKING</TYPE>
      </ERROR>
    </DEVICE>
    <MODULEVERSION>6.1</MODULEVERSION>
    <PROCESSNUMBER>189</PROCESSNUMBER>
  </CONTENT>
  <DEVICEID>TS04-2024-05-27-11-40-30</DEVICEID>
  <QUERY>SNMPQUERY</QUERY>
</REQUEST>
```

How can I fix this? Exept for manual editing the discovered port.

Regards,
Marcel

**Comment by g-bougard** [comment, marked as answer]
Hello Marcel,

this is a know problem and it is a glpi-inventory-plugin problem. You may try to open an issue on that project.

But I can propose you to better use ToolBox in place of glpi-inventory-plugin. You won't have this problem as in ToolBox device inventory is chained on each device discovery and for that discovery ip is kept.
You can check ToolBox documentation there: https://glpi-agent.readthedocs.io/en/latest/plugins/toolbox-plugin.html

Another alternative is to run [glpi-netdiscovery](https://glpi-agent.readthedocs.io/en/latest/man/glpi-netdiscovery.html) script with `-i` option to save XML files and inject them after with `glpi-injector` script.

**Reply by blackbasket** [reply]
Thanks again for your quick response @g-bougard :)

**Reply by blackbasket** [reply]
Hi @g-bougard 

I am still testing some things. I tried to use the ToolBox as advised. This works good for the devices with multiple IPs. But if I use this, there must be some other issue. All trunk ports are listed with the same MAC addresses connected. This is senseless, of course.

Do you have any advise for this?

Regards,
Marcel
